mitodl/open-discussions
# add priority for spam tasks

#### Pre-Flight checklist

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3188

#### What's this PR do?
Adds priority to celery queues so that spam tasks can be picked up before other tasks

#### How should this be manually tested?
add
```
import time
@app.task
def sleep(seconds):
    time.sleep(seconds)
```

to `course_catalog/tasks.py`

Fill the celery queues with default priority tasks by running

```
from course_catalog.tasks import *

for i in (0,1000):
    sleep.delay(15)
```

from the shell

`docker-compose run web ./manage.py reclassify_spam --spam --post-id <object id of post>`

should still run fairly quickly and if you create a post or comment with a non-moderator account a `check_post_for_spam` job should run to create a SpamCheckResult record

All other celery tasks will be queued behind the sleep jobs and will not run
